src/main/java/sirius/biz/analytics/scheduler/AnalyticalTask.java

```diff
@@ -8,29 +8,32 @@
 
 package sirius.biz.analytics.scheduler;
 
+import sirius.db.mixing.BaseEntity;
+
 import java.time.LocalDate;
 
 /**
- * Represents an analytical task which is executed for each of the entities and in the interval specified by the
- * referenced {@link AnalyticsScheduler scheduler}.
+ * Provides an analytical tasks which is executed on the given entity.
+ * <p>
+ * Subclasses (interfaces) of this need an {@link MongoAnalyticalTaskScheduler} and/or {@link SQLAnalyticalTaskScheduler}
+ * which schedules the execution of this task for all matching entities.
  *
- * @param <E> the type of entities processed by this task
+ * @param <E> the entity type to be processed by this task
  */
-public interface AnalyticalTask<E> {
+public interface AnalyticalTask<E extends BaseEntity<?>> {
 
     /**
-     * The scheduler which is used to schedules and executes this task for each of its entities.
+     * Returns the class of entities to be processed by this task.
```

Comment:
Could be removed as its the same as the `@return` line

Reply:
https://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#@return nope

> include it for all other methods, even if its content is entirely redundant with the method description. Having an explicit @return tag makes it easier for someone to find the return value quickly

Reply:
Yeah I mean the first line could be removed. Like this:
```
/**
* @return the class of entities to be processed by this task
*/